virtual_ship/instruments/ship_underwater_st.py

```diff
@@ -1,11 +1,14 @@
 """Ship salinity and temperature."""
 
 import numpy as np
-from parcels import FieldSet, JITParticle, ParticleSet, Variable
+from parcels import FieldSet, ParticleSet, ScipyParticle, Variable
 
 from ..spacetime import Spacetime
 
-_ShipSTParticle = JITParticle.add_variables(
+# we specifically use ScipyParticle because we have many small calls to execute
+# JITParticle would require compilation every time
+# this ends up being faster
```

Comment:
Interesting. Parcels should only recompile if the kernels changed, so apparently there's an issue there? Or is it just that the overhead of loading the C-libraries etc makes JIT slower even without recompilation?

Reply:
Right. I assumed it was the recompilation, but maybe its something else related. Does parcels cache the compiled kernel even if you call 'execute' multiple times?

Reply:
I'll change the comment a bit